### ISSUE TITLE
Exercise 1.21 solution in Chapter 1.2.6 gives infinite recursion

### ISSUE THREAD

**Issue by ngxingyu** [source-academy/sicp#608]
Hi, I believe the solution of https://sourceacademy.nus.edu.sg/sicpjs/1.2.6#ex-1.21 currently does not terminate since `report_prime` returns a number instead of `undefined`, so `is_undefined(timed_prime_test(start))` will always be false. Perhaps change `is_undefined` to `is_number` or change the output of `report_prime`

Error shown:
```
Line 49: Potential infinite recursion detected: search_for_primes(15303, 1) ... search_for_primes(15305, 1) ... search_for_primes(15307, 1).
```

**Comment by samuelfangjw** [comment]
Thanks for reporting this @ngxingyu.

As @ngxingyu suggested, we could change the `report_prime` function to implicitly return undefined instead.
```diff
function report_prime(elapsed_time) {
    display(" *** ");
-   return display(elapsed_time);
+   display(elapsed_time);
}
```

Full solution: https://share.sourceacademy.nus.edu.sg/cu59n

Prof @martin-henz and @ngxingyu, could you kinda take a look to see if this change solves the issue?

**Comment by ngxingyu** [comment]
Looks good to me!

**Comment by martin-henz** [comment]
Fine. @ngxingyu can you make a PR for this?

**Comment by samuelfangjw** [comment]
@ngxingyu, if you are keen or working on this issue, I'll be happy to guide you through the process if needed. Just mention me on this issue!

**Comment by ngxingyu** [comment]
Okay!

**Comment by samuelfangjw** [comment]
Closed by #609
